internal/helm/repository/chart_repository.go

```diff
@@ -70,13 +72,52 @@ type ChartRepository struct {
 	tlsConfig *tls.Config
 
 	*sync.RWMutex
+
+	cacheInfo
+}
+
+type cacheInfo struct {
+	// In memory cache of the index.yaml file.
+	IndexCache *cache.Cache
+	// IndexKey is the cache key for the index.yaml file.
+	IndexKey string
+	// IndexTTL is the cache TTL for the index.yaml file.
+	IndexTTL time.Duration
+	// RecordIndexCacheMetric records the cache hit/miss metrics for the index.yaml file.
+	RecordIndexCacheMetric RecordMetricsFunc
+}
+
+// ChartRepositoryOptions is a function that can be passed to NewChartRepository
+// to configure a ChartRepository.
+type ChartRepositoryOptions func(*ChartRepository) error
```

Comment:
```suggestion
type ChartRepositoryOption func(*ChartRepository) error
```

it's a singular as you work with a list of items.